src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,513 @@
+   ```
+
+   :::note
+   You may see a warning _"An operation was scaffolded that may result in the loss of data. Please review the migration for accuracy."_.
+   The column length for redirect URIs (for both login and logout) was reduced from 2000 to 400 to overcome database index size limits.
+   Unless you are using redirect URIs greater than 400 characters, this should not affect you.
+   :::
+
+2. Apply the migrations to your database:
+
+   ```bash title="Terminal"
+   dotnet ef database update -c PersistedGrantDbContext
+   dotnet ef database update -c ConfigurationDbContext
+   ```
+
+</Steps>
+
+### Step 6: Migrate Signing Keys <Badge text="Optional" variant="note" size="small" />
+
+If your IdentityServer4 implementation is using a signing key, consider using [automatic key management](../../fundamentals/key-management).
+
+:::tip[Determine if you are using a custom signing key]
+In `Startup.cs`, look for a call to `AddSigningCredential()` that uses key material such as an `X509Certificate2`.
+:::
+
+Client apps and APIs typically cache the key material published from IdentityServer's discovery document.
+When upgrading, consider how those applications will handle an upgraded token server with a new and different signing key.
+
+- If you can restart all client apps and APIs that depend on your current signing key, you can remove the old signing key and start to use automatic key management. A restart reloads the discovery document and the new signing key.
+- If you can not restart client apps and APIs, check the [manual and automatic key rotation topics](../../fundamentals/key-management#manual-key-rotation) to learn how to announce new signing key material while still supporting the old signing key for a period of time.
+
+### Step 7: Verify Data Protection Configuration
+
+Duende IdentityServer depends on [ASP.NET Data Protection](../../deployment#aspnet-core-data-protection) to encrypt and sign data using keys managed by ASP.NET.
+
+As part of your migration, verify the application name is set in your Data Protection configuration:
+
+```cs title="Program.cs" {4}
+builder.Services.AddDataProtection()
+  .PersistKeysTo...()
+  .ProtectKeysWith...()
+  .SetApplicationName("IdentityServer");
+```
+
+If an application name is set, you can skip this section. If no name is set, it is important to read on.
+
+Data Protection keys are isolated by application name, to prevent multiple applications from sharing encryption keys.
+If no application name is configured, ASP.NET Data Protection uses the content root path of the IdentityServer host as the application name.
+
+As a consequence, if your content root path changes, the default settings for data protection will prevent you from using your old data protection keys.
+
+.NET 6 introduced a change where the (default) content root path is now normalized to end with a directory separator.
```

Comment:
Wasn't that then reverted in .NET 7? So .NET 8-9 now have the .NET 5 behaviour?

https://github.com/dotnet/aspnetcore/issues/40964#issuecomment-1084912287